### PR TITLE
[MLIR][Presburger] Fix style violations in ff80414 (NFC)

### PR DESCRIPTION
**Pull request by iambrj**
Use preincrement not postincrement; use `Identifier::getIds` not `getVarKindOffset`

### CODE REVIEW

**Review by tobiasgrosser — APPROVED** [review]
This certainly looks a lot better. Lets see what @Groverkss and @Superty have to add.

File: mlir/lib/Analysis/Presburger/PresburgerSpace.cpp [hunk 1]
```diff
@@ -299,30 +299,25 @@ void PresburgerSpace::mergeAndAlignSymbols(PresburgerSpace &other) {
...
-       identifier++) {
+  for (const Identifier identifier : getIds(VarKind::Symbol)) {
     // If the identifier exists in `other`, then align it; otherwise insert it
     // assuming it is a new identifier. Search in `other` starting at position
     // `i` since the left of `i` is aligned.
-    auto *findEnd =
-        other.identifiers.begin() + other.getVarKindEnd(VarKind::Symbol);
-    auto *itr = std::find(other.identifiers.begin() + kindBeginOffset + i,
-                          findEnd, *identifier);
+    auto *findBegin = other.getIds(VarKind::Symbol).begin() + i;
```
Comment (Groverkss) [inline mlir/lib/Analysis/Presburger/PresburgerSpace.cpp:307]: nit: expand auto

Comment (Groverkss) [reply mlir/lib/Analysis/Presburger/PresburgerSpace.cpp:307]: I guess it's iterators so it's fine

**Review by Groverkss — APPROVED** [review]
LGTM

**Pull request merged**